smart-send-logistics/pickup-point-block/smart-send-blocks-integration.php

```diff
@@ -0,0 +1,213 @@
+<?php
+
+use Automattic\WooCommerce\Blocks\Integrations\IntegrationInterface;
+
+require_once __DIR__ . '/smart-send-extend-store-endpoint.php';
+
+/**
+ * Class for integrating with WooCommerce Blocks
+ * 
+ * This class implements the IntegrationInterface to extend the WooCommerce
+ * Store API, register custom scripts and styles, and save custom shipping
+ * instructions.
+ * 
+ * @package    SmartSendLogistics
+ * @subpackage API Extension
+ * @category   WooCommerce Blocks IntegrationInterface implements
+ * @author     Smart Send
+ */
+
+
+class Smart_Send_Blocks_Integration implements IntegrationInterface
+{
+
+    /**
+     * Gets the name of the integration.
+     *
+     * @return string
+     */
+    public function get_name()
+    {
+        return SS_SHIPPING_WOO_BLOCK_NAME;
+    }
+
+    /**
+     * Initializes the integration by registering scripts, styles, and extending the store API.
+     */
+    public function initialize()
+    {
+        $this->register_smart_send_block_frontend_scripts();
+        $this->register_smart_send_block_editor_scripts();
+        $this->register_smart_send_block_editor_styles();
+        $this->register_main_integration();
+        $this->extend_store_api();
+        $this->save_shipping_instructions();
+    }
+
+    /**
+     * Extends the cart schema to include the smart-send value.
+     */
+    private function extend_store_api()
+    {
+        Smart_Send_Extend_Store_Endpoint::init();
+    }
+
+    private function save_shipping_instructions()
+    {
+        add_action(
+            'woocommerce_store_api_checkout_update_order_from_request',
+            function (\WC_Order $order, \WP_REST_Request $request) {
+                $smart_send_request_data = $request['extensions'][$this->get_name()];
+                $pickup_points = $smart_send_request_data['selectedPickupPoint'];
```

Comment:
Where is this string `'selectedPickupPoint'` coming from? Can we please define this as a constant and use it like that.

Reply:
The string  `selectedPickupPoint `is  basically the key name  from the array which store our blocks data and in above mentioned code we are extracting it from the checkout api...
yes this can be a constant.

Reply:
Where is the structure of `$smart_send_request_data` defined?

It would be better converting `$smart_send_request_data` into a DTO.